fetch_data.py

```diff
@@ -419,25 +438,72 @@ def create_csv(packaged_data, fieldnames, package_coords, fill_di=None):
                 row_di[fieldname] = value
         # write the actual value
         row_di["value"] = get_from_dict(packaged_data, coords)
-        writer.writerow(row_di)
-    return output.getvalue()
+        rows.append(row_di)
+    return rows
 
 
-def return_csv(csv_data):
-    """Return the CSV data as a download
+def write_csv(csv_dicts, fieldnames, filename, metadata=None):
+    """
+    Creates and returns a downloadable CSV file from list of CSV dicts.
 
     Args:
-        csv_data (?): csv data created with create_csv() function
+        csv_dicts (list): CSV data created with build_csv_dicts function.
+        fieldnames (list): list of fields used to create CSV header row
+        filename (str): File name of downloaded CSV file
 
     Returns:
         CSV Response
     """
+    output = io.StringIO()
+    if metadata is not None:
+        output.write(metadata)
+    writer = csv.DictWriter(output, fieldnames=fieldnames)
+    writer.writeheader()
+    writer.writerows(csv_dicts)
     response = Response(
-        csv_data,
+        output.getvalue(),
         mimetype="text/csv",
         headers={
-            "Content-Type": 'text/csv; name="climate.csv"',
-            "Content-Disposition": 'attachment; filename="climate.csv"',
+            "Content-Type": "text/csv; charset=utf-8",
+            "Content-Disposition": 'attachment; filename="'
+            + filename
+            + "\"; filename*=utf-8''\""
+            + filename
+            + '"',
         },
     )
     return response
+
+
+def csv_metadata(place_name, place_id, place_type, lat=None, lon=None):
+    """
+    Creates metadata string to add to beginning of CSV file.
+
+    Args:
+        place_name (str): Name of the place, None of just lat/lon
+        place_id (str): place identifier (e.g., AK124)
+        place_type (str): point or area
+        lat: latitude for points or None for polygons
+        lon: longitude for points or None for polygons
+
+    Returns:
+        Multiline metadata string
+    """
+    metadata = "# Location: "
+    if place_name is None:
+        metadata += lat + " " + lon + "\n"
+    elif place_type == "communities":
+        metadata += place_name + "\n"
+    else:
+        metadata += place_name + " (" + place_type_labels[place_type] + ")\n"
+
+    report_url = WEB_APP_URL + "report/"
```

Comment:
OK I just realized now that we *probably* want to move this code into a string that is provided to the API, this way the API doesn't need any know anything about the URL structure of the client app.  (This isn't wrong, it's just nice to keep them as separate as possible).